modules/core/src/lib/deck.js

```diff
@@ -445,21 +440,16 @@ export default class Deck {
 
     this.stats.bump('render-fps');
 
-    if (this.props.onBeforeRender) {
-      this.props.onBeforeRender({gl}); // TODO - should be called by AnimationLoop
-    }
-    this.layerManager.drawLayers({
-      pass: 'screen',
-      redrawReason,
-      // Helps debug layer picking, especially in framebuffer powered layers
-      drawPickingColors: this.props.drawPickingColors
-    });
-    if (this.props.onAfterRender) {
-      this.props.onAfterRender({gl}); // TODO - should be called by AnimationLoop
-    }
+    setParameters(gl, this.props.parameters);
```

Comment:
Does this have any perf implication? Do we need to only apply it if the prop has changed?

Reply:
> Does this have any perf implication? Do we need to only apply it if the prop has changed?

luma.gl `setParams` already does exactly that.